configs/components/facter.rb

```diff
@@ -219,6 +219,14 @@
     ["#{make} -j$(shell expr $(shell #{platform[:num_cores]}) + 1) install"]
   end
 
+  pkg.check do
+    [
+      # Ensure we're linking against our libstdc++ and libgcc_s or statically linking
+      "ldd lib/libfactero.so | grep libstdc++ && ldd lib/libfacter.so | grep libstdc++ | grep puppet || true",
+      "ldd lib/libfacter.so | grep libgcc_s && ldd lib/libfacter.so | grep puppet || true"
```

Comment:
Tested against: sles-12-s390x, solaris-10-i386, solaris-11-sparc, el-5-i386, el-7-x86_64 and ubuntu-16.04-amd64 without issues.